arroyo-worker/src/operators/join_with_expiration.rs

```diff
@@ -21,299 +25,506 @@ pub struct JoinWithExpiration<
     left_expiration: Duration,
     right_expiration: Duration,
     processor: P,
-    _t: PhantomData<(K, T1, T2, Output)>,
+    pub _t: PhantomData<(K, P1, P2, InT1, InT2, T1, T2, Output)>,
```

Comment:
I think it'd be a cleaner to just have a `new()` method that other parts of the code call, instead of making this pub or pub (crate)